quandl/model/dataset.py

```diff
@@ -12,12 +12,25 @@ class Dataset(GetOperation, ListOperation, ModelBase):
     def get_path(cls):
         return "%s/metadata" % cls.default_path()
 
+    @classmethod
+    def get_code_from_meta(cls, metadata):
+        return "%s/%s" % (metadata['database_code'], metadata['dataset_code'])
+
+    def __init__(self, code, raw_data=None, **options):
+        ModelBase.__init__(self, code, raw_data)
+        self.database_code = self.code.split("/")[0]
+        self.dataset_code = self.code.split("/")[1]
```

Comment:
`BadRequestError` and maybe an example of the valid format? or point to the docs.